test/integ/pandas/test_pandas_tools.py

```diff
@@ -300,6 +300,21 @@ def test_invalid_table_type_write_pandas(
             )
 
 
+def test_empty_dataframe_write_pandas(
+    conn_cnx: Callable[..., Generator[SnowflakeConnection, None, None]],
+):
+    table_name = random_string(5, "empty_dataframe_")
+    df = pandas.DataFrame([], columns=["name", "balance"])
+    with conn_cnx() as cnx:
+        try:
+            success, num_chunks, num_rows, _ = write_pandas(
+                cnx, df, table_name, auto_create_table=True
```

Comment:
We could use a temp table for this:
```suggestion
                cnx, df, table_name, auto_create_table=True, table_type="temp",
```